### PR TITLE
SALTO-784 Support KnowledgeSettings

### PR DESCRIPTION
**Pull request by nitzan-haim**

### CODE REVIEW

File: packages/salesforce-adapter/src/transformers/transformer.ts [hunk 1]
```diff
@@ -1341,6 +1370,10 @@ export const createMetadataTypeElements = async ({
   const enrichedFields = await Promise.all(fields.map(async field => {
     if (shouldEnrichFieldValue(field)) {
       const innerFields = await client.describeMetadataType(field.soapType)
+      if (soapTypesMissingFields[field.soapType]) {
```
Comment (adamro) [inline packages/salesforce-adapter/src/transformers/transformer.ts:1373]: nit: prefer using explicit undefined checks if that's what you mean
 `if (soapTypesMissingFields[field.soapType] !== undefined) {`

File: packages/salesforce-adapter/src/transformers/transformer.ts [hunk 2]
```diff
@@ -721,6 +721,35 @@ export class Types {
...
+  minOccurs: 0,
+  name: 'field',
+  soapType: '',
+  valueRequired: true,
+  fields: [],
+  picklistValues: [],
+}
+
+// those soapTypes are missing fields when returned from client.describeMetadataType
+const soapTypesMissingFields: Record<string, ValueTypeField> = { // <soapType, field-to-add>
```
Comment (adamro) [inline packages/salesforce-adapter/src/transformers/transformer.ts:737]: There isn't a way to get this information from the api and not it being hard coded? 
This is quite risky for changes in the service and for information that we currently don't use but might make use of in the future
I think it should be a last resort, is it?

**Review by adamro — COMMENTED** [review]
Should probably add a test for this, can maybe even make sense to add an e2e for it I guess

File: packages/salesforce-adapter/src/transformers/transformer.ts (hunk 2 above)
Comment (nitzan-haim) [reply packages/salesforce-adapter/src/transformers/transformer.ts:737]: unfortunately we can't get this information from the api, it's salesforce's bug that those fields are missing..

Comment (ori-moisis) [reply packages/salesforce-adapter/src/transformers/transformer.ts:737]: we have a missing fields filter to handle missing fields, is there a reason this PR creates a new mechanism for handling missing fields?

**Review by ori-moisis — APPROVED** [review]
LGTM

**Pull request merged**